src/scenic/syntax/translator.py

```diff
@@ -985,19 +1029,32 @@ def wrapStatementCall():
 					inConstructorContext = True
 					allowedPrefixOps = self.specifiersForConstructor(context)
 				else:
-					for opTokens, op in infixTokens.items():
-						if not op.contexts or context in op.contexts:
-							allowedInfixOps[opTokens] = op.token
 					for name, mod in modifierNames.items():
 						if not mod.contexts or context in mod.contexts:
 							allowedModifiers[name] = mod.name
 					if context in modifierNames:
 						allowedTerminators = modifierNames[context].terminators
 					elif context in terminatorsForStatements:
 						allowedTerminators = terminatorsForStatements[context]
+			# construct allowedInfixOps
+			if context:
+				# all function calls on the stack
+				contexts = set(ctx for ctx, _ in functionStack)
+				for opTokens, op in infixTokens.items():
+					# if not context is specified, the operator can be used
+					if not op.contexts:
+						allowedInfixOps[opTokens] = op.tokens
+						continue
+					# if operator is start level only and it is start level, cannot be used
+					if op.startLevelOnly and parenLevel != startLevel:
+						continue
+					# a set of contexts under which this operator is allowed
+					allowedContexts = set(op.contexts)
```

Comment:
Can we create these sets once at operator definition time to avoid constantly recreating them? It might be faster to use a `frozenset` too (although not worth spending much time investigating that).

Reply:
I agree that these sets should be created once and reused. Although `op.contexts` was a tuple, the order should not matter. Therefore, I changed the type of `contexts` field of `InfixOp` to `FrozenSet` so we can just take the intersection.

I found [this StackOverflow post](https://stackoverflow.com/questions/36555214/set-vs-frozenset-performance) which claims that there's no big performance difference between `set` and `frozenset`. However, these sets clearly should be immutable and I believe it makes sense to use `frozenset`.